lib/ramble/ramble/expander.py

```diff
@@ -906,24 +906,41 @@ def _eval_unary_ops(self, node):
         except KeyError:
             raise SyntaxError("Unsupported unary operator")
 
-    def _eval_susbscript_op(self, node):
+    def _eval_subscript_op(self, node):
         """Evaluate subscript operation in the ast"""
         try:
             operand = self.eval_math(node.value)
             slice_node = node.slice
-            if not isinstance(operand, str) or not isinstance(slice_node, ast.Slice):
-                raise SyntaxError("Currently only string slicing is supported for subscript")
-
-            def _get_with_default(s_node, attr, default):
-                v_node = getattr(s_node, attr)
-                if v_node is None:
-                    return default
-                return self.eval_math(v_node)
-
-            lower = _get_with_default(slice_node, "lower", 0)
-            upper = _get_with_default(slice_node, "upper", len(operand))
-            step = _get_with_default(slice_node, "step", 1)
-            return operand[slice(lower, upper, step)]
+
+            if isinstance(operand, str):
+                if isinstance(slice_node, ast.Slice):
+
+                    def _get_with_default(s_node, attr, default):
+                        v_node = getattr(s_node, attr)
+                        if v_node is None:
+                            return default
+                        return self.eval_math(v_node)
+
+                    lower = _get_with_default(slice_node, "lower", 0)
+                    upper = _get_with_default(slice_node, "upper", len(operand))
+                    step = _get_with_default(slice_node, "step", 1)
+                    return operand[slice(lower, upper, step)]
+                elif operand in self._variables and isinstance(self._variables[operand], dict):
+                    op_dict = self.expand_var_name(operand, typed=True)
+
+                    # TODO: Remove after support for python 3.9 is dropped
+                    # DEPRECATED: ast.Index was dropped in python 3.9
+                    if hasattr(ast, "Index") and isinstance(slice_node, ast.Index):
+                        key = self.eval_math(slice_node.value)
+                        return op_dict[key]
```

Comment:
What happens when key is not in the dict?

Reply:
Ah good catch. Currently, you get a `KeyError`, but I don't think that's what we want.

I'll work on fixing this.

Reply:
Fixed!